test/parallel/test-fs-readfile-tostring-fail.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const kMaxLength = process.binding('buffer').kMaxLength;
+
+common.refreshTmpDir();
+
+const file = path.join(common.tmpDir, 'toobig.txt');
+const stream = fs.createWriteStream(file, {
+  flags: 'a'
+});
+
+const size = 269 * 1024 * 1024;
+
+const a = new Array(1000000).join('a').toString();
```

Comment:
The operation itself is faster, but operating on that string is slower.
```
'a'.repeat(1000000):                    4.5 us/op
Buffer(1000000).fill('a').toString(): 180.8 us/op
```
Now turn those back into a Buffer to be written to disk:
```
Buffer('a'.repeat(1000000)):          7060.1 us/op
Buffer(''+Buffer(1000000).fill('a')):  721.8 us/op
```
Though there may be some bias from running these in a loop. So overall test time would be the best measurement.